### PR TITLE
Refactor controllers

### PR DESCRIPTION
**Pull request by Dantemss**
Pull in refactored controllers, routines, handlers, routes

### CODE REVIEW

File: app/assets/javascripts/application/ui.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+(function() {
```
Comment (mwvolo) [inline app/assets/javascripts/application/ui.js:1]: do we need both `application/ui.js` and `application/accounts-ui.js`?

File: app/assets/javascripts/multiselect/index.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
-//= require knockout
+//= require knockout.js
```
Comment (mwvolo) [inline app/assets/javascripts/multiselect/index.js:1]: when does something need .js on the end vs not?

File: app/assets/javascripts/profile/index.js [hunk 3]
```diff
@@ -1,16 +1,18 @@
-//= require bootstrap-editable
-//= require ../vendor/underscore
+//= require jquery
+//= require jquery_ujs
+//= require bootstrap-sprockets
+//= require underscore.js
+//= require x-editable-bootstrap.js
```
Comment (mwvolo) [inline app/assets/javascripts/profile/index.js:5]: any reason for the change to bootstrap-editable then back to x-editable-bootstrap?

**Comment by mwvolo** [comment]
ah, I started reviewing - but I think the merge of your other PR created conflicts, so I'll wait until those are resolved @Dantemss

File: app/assets/javascripts/multiselect/index.js (hunk 2 above)
Comment (Dantemss) [reply app/assets/javascripts/multiselect/index.js:1]: I don't think it needs this? Likely the PR I pulled it from just had it

**Comment by Dantemss** [comment]
The conflicts are unexpected, since it was based on that PR... oh maybe it's because of the squash merge...

File: app/assets/javascripts/profile/index.js (hunk 3 above)
Comment (Dantemss) [reply app/assets/javascripts/profile/index.js]: I think this change is gone after the rebase

File: app/assets/javascripts/application/ui.js (hunk 1 above)
Comment (Dantemss) [reply app/assets/javascripts/application/ui.js]: They do seem very similar, I imagine they could be combined... I might leave that one for you though :)

**Comment by Dantemss** [comment]
Maybe the sheerid webhook thing should be in some sort of API controller, since it's basically an API call...

**Comment by Dantemss** [comment]
@mwvolo conflicts fixed

**Comment by Dantemss** [comment]
Oh guess I should say the educator/student signup route changes might cause some errors if someone is in the middle of that process when deployed.

Maybe we need to add nginx rules for those. Or do it early morning or something.

File: app/controllers/login_controller.rb [hunk 4]
```diff
@@ -23,9 +25,9 @@ def login
           save_unverified_user(user.id)
 
           if user.student?
```
Comment (mwvolo) [inline app/controllers/login_controller.rb:27]: no need to check user role here? just `redirect_to(verify_email_by_pin_form_path)`

File: app/handlers/create_password.rb [hunk 5]
```diff
@@ -1,26 +1,34 @@
+# Changes the user's password if they have one, otherwise creates a password (an `Identity`).
 class CreatePassword
   lev_handler
-  uses_routine ::SetPassword
+  uses_routine SetPassword
```
Comment (mwvolo) [inline app/handlers/create_password.rb:4]: why remove `::` from here but not line 23? Does `::` do something special?

File: config/initializers/assets.rb [hunk 6]
```diff
@@ -8,6 +8,7 @@
 Rails.application.config.assets.precompile += %w(
```
Comment (mwvolo) [inline config/initializers/assets.rb:8]: do we still need this file with https://github.com/openstax/accounts/blob/refactor_controllers/app/assets/config/manifest.js?

File: config/routes.rb [hunk 7]
```diff
@@ -7,125 +6,111 @@
...
-
   # routes for access via an iframe
   scope 'remote', controller: 'remote' do
     get 'iframe'
     get 'notify_logout', as: 'iframe_after_logout'
   end
 
-  scope controller: 'users' do
+  scope controller: :users do
     put 'profile', action: :update
```
Comment (mwvolo) [inline config/routes.rb:110]: I think as part of another PR, this should either move to the profile_controller or the profile "show/edit" view should go the users controllers.

**Review by mwvolo — APPROVED** [review]
Left a few comments/questions - but overall excellent 👏🏼 💯 
Thanks so much for getting this into a working state, @Dantemss!

File: app/handlers/create_password.rb (hunk 5 above)
Comment (Dantemss) [reply app/handlers/create_password.rb:4]: The `::` ensures the SetPassword being called is the top-level one, not SomeModule::SetPassword.
It's not necessary unless you have multiple SetPassword or think you may have them in the future. I can probably remove all of them.

File: config/initializers/assets.rb (hunk 6 above)
Comment (Dantemss) [reply config/initializers/assets.rb:8]: The manifest is used by sprockets 4+ but we still have sprockets 3 in the Gemfile

**Pull request merged**